source/distributed/tria_base.cc

```diff
@@ -429,11 +432,17 @@ namespace parallel
             continue;
 
           if (cell->is_locally_owned())
-            is_local.add_index(index);
+            is_local_vector.push_back(index);
           else
-            is_ghost.add_index(index);
+            is_ghost_vector.push_back(index);
         }
 
+    std::sort(is_local_vector.begin(), is_local_vector.end());
+    is_local.add_indices(is_local_vector.begin(), is_local_vector.end());
```

Comment:
One thing I see now: Would you mind moving the declaration of the `IndexSet` variables here to their first use?

Reply:
Done.